### DISCUSSION TITLE
Two methods of deployment Azure Data Factory

### DISCUSSION THREAD

**Discussion by NowinskiK** [AzureCommunityPL/group#200] [Community Content]
Czołem koledzy!

Dawno nie było moich aktywności blogowych, ale pierwszy post w tym roku zobowiązuje i powiem nieskromnie: "jest moc". 
Materiały do tego posta zbierałem od kilku miesięcy, zresztą sam pomysł na posta powstał już prawie rok temu, a od tamtego czasu byłem pytany przez wiele osób "jak to robić?". 
O czym rzecz? **Azure Data Factory**, a dokładniej: **wdrażanie (deployment) ADF na środowiska docelowe**.
Są dwie szkoły: podejście Microsoft oraz publikacja z pominięciem ARM Templates (moje ulubione).
Jeśli kiedykolwiek zastanawialiście się nad wadami i zaletami obu rozwiązań, albo znacie tylko jedno z nich, albo (co gorsza) nadal sprawne wdrażanie kodu Azure Data Factory stanowi problem - zapraszam do zapoznania się z tym postem:
🌐 Post: [Two methods of deployment Azure Data Factory](https://sqlplayer.net/2021/01/two-methods-of-deployment-azure-data-factory/)

![image](https://user-images.githubusercontent.com/25029325/104811122-4453a780-57f1-11eb-9513-018cfff3baaf.png)

Zachęcam również do dyskusji i podzielenia się Waszymi doświadczeniami z "pola walki".
Udanej lektury!

**Comment by JanuszNowak** [comment]
Masz jeszcze trzecią metodę oprócz repozytorium git czy arm templatki,  przez Azure Power shella i pliki json

**Reply by NowinskiK** [reply]
To jest właśnie ta druga metoda.

**Comment by szympulka** [comment]
Mam takie samo flow jak Ty zaimplementowane i przy wrzuceniu wersji na inne środowiska dostaję błąd. Cannot update enable Trigger. Ręczne wyłączenie triggera wystarczyło, lecz jest to męczące na dłuższą mete. Jak sobie z tym radzisz?
![MicrosoftTeams-image (3)](https://user-images.githubusercontent.com/12598823/105728595-a072a600-5f2c-11eb-89e2-a8e5f295f100.png)

**Reply by NowinskiK** [reply]
Czym to robisz? 
Mój moduł wyłącza wszystkie (włączone) triggery na środowisku docelowym przed rozpoczęciem publikowania obiektów (zgodnie zresztą z tym co zaleca MS).
Jeśli to zautomatyzujesz to przestaje być męczące i przestajesz o tym myśleć.

**Reply by szympulka** [reply]
Leci Terraform, nie grzebałem tam jeszcze ale będę szukać skoro jest taka możliowść
